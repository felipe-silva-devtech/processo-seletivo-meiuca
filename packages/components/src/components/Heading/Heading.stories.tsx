import React from "react";

import { StoryFn, Meta } from "@storybook/react";

import { Heading, HeadingProps } from ".";

export default {
  title: "Components / Heading",
  component: Heading,
  argTypes: {
    content: { control: { type: "text" } },
  },
  args: {
    content: "Heading SM",
  },
} as Meta<HeadingProps>;

export const Default: StoryFn<typeof Heading> = (args) => <Heading {...args} />;
