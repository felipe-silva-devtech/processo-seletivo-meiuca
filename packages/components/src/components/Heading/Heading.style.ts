import styled from "@emotion/styled";
import { basicProps } from "../../utils/styled-system-utils";

export const StyledHeading = styled("h1")`
  ${basicProps}

  font-size: ${({ theme }) => theme.fontSizes.lg};
  line-height: ${({ theme }) => theme.lineHeights.distant};
  font-family: ${({ theme }) => theme.fonts.highlight};
  font-weight: ${({ theme }) => theme.fontWeights.bold};
  color: ${({ theme }) => theme.colors.neutral["01"]};
`;
