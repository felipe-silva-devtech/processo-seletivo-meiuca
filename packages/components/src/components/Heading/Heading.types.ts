import React from "react";

export type HeadingProps = React.PropsWithChildren & {
  content?: string;
};
