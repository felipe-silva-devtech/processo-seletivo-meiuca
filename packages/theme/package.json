{
  "name": "@psm/theme",
  "version": "1.0.0",
  "private": true,
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "license": "MIT",
  "scripts": {
    "prebuild": "rm -rf dist",
    "build": "rollup -c"
  },
  "dependencies": {
    "@emotion/react": "^11.11.0"
  },
  "devDependencies": {
    "@rollup/plugin-commonjs": "^25.0.0",
    "@rollup/plugin-typescript": "^11.1.1",
    "rollup": "^2.79.1"
  }
}
