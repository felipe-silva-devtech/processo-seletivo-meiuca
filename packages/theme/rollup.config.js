import commonjs from "@rollup/plugin-commonjs";
import typescript from "@rollup/plugin-typescript";

const packageJson = require("./package.json");

export default {
  input: "src/index.ts",
  cache: false,
  output: [
    {
      file: packageJson.main,
      format: "cjs",
    },
    {
      file: packageJson.module,
      format: "esm",
    },
  ],
  plugins: [commonjs(), typescript({ tsconfig: "./tsconfig.json" })],
};
