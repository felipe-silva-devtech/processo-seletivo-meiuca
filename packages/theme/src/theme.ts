import tokens from "@psm/tokens";

export const theme = {
  lineHeights: {
    tight: tokens.lineHeightTight,
    medium: tokens.lineHeightMedium,
    distant: tokens.lineHeightDistant,
    superdistant: tokens.lineHeightSuperdistant,
  },
  fontSizes: {
    xxxs: tokens.fontSizeXxxs,
    xxs: tokens.fontSizeXxs,
    xs: tokens.fontSizeXs,
    sm: tokens.fontSizeSm,
    md: tokens.fontSizeMd,
    lg: tokens.fontSizeLg,
    xl: tokens.fontSizeXl,
    xxl: tokens.fontSizeXxl,
    xxxl: tokens.fontSizeXxxl,
    display: tokens.fontSizeDisplay,
    giant: tokens.fontSizeGiant,
  },
  radii: {
    none: tokens.borderRadiusNone,
    sm: tokens.borderRadiusSm,
    md: tokens.borderRadiusMd,
    lg: tokens.borderRadiusLg,
    pill: tokens.borderRadiusPill,
    circular: tokens.borderRadiusCircular,
  },
  borderWidths: {
    none: tokens.borderWidthNone,
    hairline: tokens.borderWidthHairline,
    thin: tokens.borderWidthThin,
    thick: tokens.borderWidthThick,
    heavy: tokens.borderWidthHeavy,
  },
  opacity: {
    semiopaque: tokens.opacityLevelSemiopaque,
    intense: tokens.opacityLevelIntense,
    medium: tokens.opacityLevelMedium,
    light: tokens.opacityLevelLight,
    semitransparent: tokens.opacityLevelSemitransparent,
  },
  shadows: {
    "01": tokens.shadowLevel1Stack,
    "02": tokens.shadowLevel2Stack,
    "03": tokens.shadowLevel3Stack,
    "04": tokens.shadowLevel4Stack,
  },
  space: {
    stack: {
      quarck: tokens.spacingStackQuarck,
      nano: tokens.spacingStackNano,
      xxxs: tokens.spacingStackXxxs,
      xxs: tokens.spacingStackXxs,
      xs: tokens.spacingStackXs,
      sm: tokens.spacingStackSm,
      md: tokens.spacingStackMd,
      lg: tokens.spacingStackLg,
      xl: tokens.spacingStackXl,
      xxl: tokens.spacingStackXxl,
      xxxl: tokens.spacingStackXxxl,
      huge: tokens.spacingStackHuge,
      giant: tokens.spacingStackGiant,
    },
    inset: {
      quarck: tokens.spacingInsetQuarck,
      nano: tokens.spacingInsetNano,
      xs: tokens.spacingInsetXs,
      sm: tokens.spacingInsetSm,
      md: tokens.spacingInsetMd,
      lg: tokens.spacingInsetLg,
    },
    squish: {
      quarck: tokens.spacingSquishQuarckStack,
      nano: tokens.spacingSquishNanoStack,
      xs: tokens.spacingSquishXsStack,
      sm: tokens.spacingSquishSmStack,
    },
    inline: {
      quarck: tokens.spacingInlineQuarck,
      nano: tokens.spacingInlineNano,
      xxxs: tokens.spacingInlineXxxs,
      xxs: tokens.spacingInlineXxs,
      xs: tokens.spacingInlineXs,
      sm: tokens.spacingInlineSm,
      md: tokens.spacingInlineMd,
      lg: tokens.spacingInlineLg,
      xl: tokens.spacingInlineXl,
    },
  },
  fonts: {
    highlight: tokens.fontFamilyHighlight,
  },
  fontWeights: {
    bold: tokens.fontWeightBold,
    medium: tokens.fontWeightMedium,
    regular: tokens.fontWeightRegular,
  },
  colors: {
    neutral: {
      "01": tokens.colorNeutral1,
      "02": tokens.colorNeutral2,
      "03": tokens.colorNeutral3,
      "04": tokens.colorNeutral4,
      "05": tokens.colorNeutral5,
    },
    brand: {
      primary: {
        "01": tokens.colorBrandPrimary1,
        "02": tokens.colorBrandPrimary2,
        "03": tokens.colorBrandPrimary3,
        "04": tokens.colorBrandPrimary4,
        "05": tokens.colorBrandPrimary5,
      },
    },
  },
};

export type Theme = typeof theme;
