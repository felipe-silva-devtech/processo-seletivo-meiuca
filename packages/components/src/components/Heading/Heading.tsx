import React from "react";

import { StyledHeading } from "./Heading.style";
import { HeadingProps } from "./Heading.types";

export const Heading: React.FC<HeadingProps> = ({
  children,
  content = "",
  ...props
}) => {
  return (
    <StyledHeading {...props}>{children ? children : content}</StyledHeading>
  );
};
