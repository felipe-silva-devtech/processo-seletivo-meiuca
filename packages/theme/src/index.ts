import { ThemeProvider } from "./provider";
import { theme, Theme } from "./theme";

export { ThemeProvider, theme };
export type { Theme };
