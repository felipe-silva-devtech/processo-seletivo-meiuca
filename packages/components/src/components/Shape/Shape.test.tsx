import React from "react";
import { render, RenderOptions } from "@testing-library/react";
import "@testing-library/jest-dom";

import { ThemeProvider, theme } from "@psm/theme";
import { Shape } from ".";
import { Heading } from "../Heading";

function renderWithWrapper(
  component?: React.ReactElement,
  options?: RenderOptions
) {
  const Wrapper = <ThemeProvider theme={theme}>{component}</ThemeProvider>;

  return render(Wrapper, options);
}

describe("Shape", () => {
  const content = "Heading Label";

  it("render component", () => {
    const { container, getByText } = renderWithWrapper(
      <Shape>
        <Heading>{content}</Heading>
      </Shape>
    );

    expect(getByText(content)).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  it("render component with custom width and height", () => {
    const { container, getByText } = renderWithWrapper(
      <Shape width={200} height={200}>
        <Heading>{content}</Heading>
      </Shape>
    );

    expect(getByText(content)).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
