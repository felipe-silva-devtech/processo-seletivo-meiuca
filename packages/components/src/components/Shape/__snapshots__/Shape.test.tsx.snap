// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shape render component 1`] = `
<div>
  <div
    class="css-103z0u2"
  >
    <h1
      class="css-1j1wjfe"
    >
      Heading Label
    </h1>
  </div>
</div>
`;

exports[`Shape render component with custom width and height 1`] = `
<div>
  <div
    class="css-l313m4"
    height="200"
    width="200"
  >
    <h1
      class="css-1j1wjfe"
    >
      Heading Label
    </h1>
  </div>
</div>
`;
